mesmer/stats/_auto_regression.py

```diff
@@ -595,3 +601,339 @@ def _fit_auto_regression_np(data, lags):
+    _check_dataset_form(ar_params, "ar_params", required_vars=("intercept", "slope"))
+
+    (month_dim, gridcell_dim) = ar_params.intercept.dims
+    (n_months, n_gridpoints) = ar_params.intercept.shape
+
+    _check_dataarray_form(
+        ar_params.intercept,
+        "intercept",
+        ndim=2,
+        required_dims=(month_dim, gridcell_dim),
+    )
+    _check_dataarray_form(
+        ar_params.slope, "slope", ndim=2, required_dims=(month_dim, gridcell_dim)
+    )
+    result = _draw_ar_corr_monthly_xr_internal(
+        intercept=ar_params.intercept,
+        slope=ar_params.slope,
+        covariance=None,
+        time=time,
+        realisation=1,
+        seed=0,
+        buffer=buffer,
+        time_dim="time",
+        realisation_dim="realisation",
+    ).squeeze("realisation", drop=True)
+
+    return result
+
+
+def draw_auto_regression_monthly(
+    ar_params,
+    covariance,
+    time,
+    n_realisations,
+    seed,
+    buffer,
+    time_dim="time",
+    realisation_dim="realisation",
+):
+    """draw time series of an auto regression process with lag one
+    using individual parameters for each month including spatially-correlated innovations.
+
+    Parameters
+    ----------
+    ar_params : xr.Dataset
+        Dataset containing the estimated parameters of the AR1 process. Must contain the
+        following DataArray objects:
+
+        - intercept
+        - slope
+
+        both of shape (12, n_gridpoints).
+    covariance : xr.DataArray of shape (12, n_gridpoints, n_gridpoints)
+        The covariance matrix representing spatial correlation between gridpoints for
+        each month. Must be symmetric and at least positive-semidefinite.
+        Used to draw spatially-correlated innovations using a multivariate normal.
+    time : xr.DataArray
+        The time coordinates that determines the length of the predicted timeseries and
+        that will be the assigned time dimension of the predictions.
+    n_realisations : int
+        The number of realisations to draw.
+    seed : int
+        Seed used to initialize the pseudo-random number generator.
+    buffer : int
+        Buffer to initialize the autoregressive process (ensures that start at 0 does
+        not influence overall result).
+    time_dim : str, default "time"
+        Name of the time dimension for the output data.
+    realisation_dim : str, default "realisation"
+        Name of the realisation dimension for the output data.
+
+    Returns
+    -------
+    result : xr.DataArray of shape (n_realisations, n_timesteps, n_gridpoints)
+        Predicted time series of the specified AR(1) process including spatially
+        correlated innovations. The array has shape n_timesteps x n_gridpoints.
+
+    """
+    # check input
+    _check_dataset_form(ar_params, "ar_params", required_vars=("intercept", "slope"))
+    (month_dim, gridcell_dim), (n_months, size) = (
+        ar_params.intercept.dims,
+        ar_params.intercept.shape,
+    )
```

Comment:
```suggestion
    month_dim, gridcell_dim = ar_params.intercept.dims,
    n_months, size = ar_params.intercept.shape
```

Nested tuples - cool :wink: Good idea but as it leads to more lines of code, I would not do it

Reply:
Agh, mean 😁 but makes sense oc.